src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int MAX_CAPACITY = 10;
+    private static final float GROWTH_COEFFICIENT = 1.5f;
+    private static final String NO_ELEMENT = "No such element in the list";
+    private static final String INDEX_ERROR = "The index is out of list`s interval";
+    private Object[] data;
+    private int size;
 
+    public ArrayList() {
+        data = new Object[MAX_CAPACITY];
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T value) {
+        if (size == data.length) {
+            increaseCapacity();
+        }
+        data[size++] = value;
+    }
 
+    @Override
+    public void add(T value, int index) throws ArrayListIndexOutOfBoundsException {
```

Comment:
Exception is RuntimeException, you don't need use throws
```suggestion
    public void add(T value, int index) {
```